src/test/java/org/jenkinsci/plugins/ewm/casc/ConfigAsCodeTest.java

```diff
@@ -0,0 +1,92 @@
+package org.jenkinsci.plugins.ewm.casc;
+
+import hudson.ExtensionList;
+import jenkins.model.Jenkins;
+import org.jenkinsci.plugins.ewm.definitions.Disk;
+import org.jenkinsci.plugins.ewm.definitions.DiskPool;
+import org.jenkinsci.plugins.ewm.steps.ExwsAllocateStep;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.*;
+
+import org.jvnet.hudson.test.JenkinsRule;
+import io.jenkins.plugins.casc.ConfigurationAsCode;
+import org.yaml.snakeyaml.Yaml;
+
+import java.io.*;
+import java.net.URL;
+import java.nio.file.Files;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Test for Configuration As Code Compatibility.
+ *
+ * @author Martin d'Anjou
+ */
+public class ConfigAsCodeTest {
+
+    @ClassRule public static JenkinsRule r = new JenkinsRule();
+
+    @Test
+    public void should_support_configuration_as_code() throws Exception {
+        URL resource = ConfigAsCodeTest.class.getResource("configuration-as-code.yaml");
+        String config = resource.toString();
+        ConfigurationAsCode.get().configure(config);
+        // parse the yaml file and check if they match
+        File file  = new File(resource.getFile());
+        String configContent = new String(Files.readAllBytes(file.toPath()));
+        Yaml yaml = new Yaml();
+        Map<String, Object> map = (Map<String, Object>) yaml.load(configContent);
+        // get the jenkins rule object.
+
+        ExwsAllocateStep.DescriptorImpl descriptor =  ExtensionList.lookupSingleton(ExwsAllocateStep.DescriptorImpl.class);
+        List<DiskPool> diskPools = descriptor.getDiskPools();
+        DiskPool diskPool = diskPools.get(0);
+
+        // assertion
+        assertThat(diskPool.getDiskPoolId(), is("diskpool1"));
+        assertThat(diskPool.getDisplayName(), is("diskpool1 display name"));
+        assertThat(diskPool.getDisks().get(0).getDiskId(), is("disk1"));
+        assertThat(diskPool.getDisks().get(0).getDisplayName(), is("disk one display name"));
+        assertThat(diskPool.getDisks().get(0).getMasterMountPoint(), is("/tmp"));
+    }
+
+    @Test
+    public void should_be_backward_compatible() throws Exception {
+        String config = ConfigAsCodeTest.class.getResource("obsolete-configuration-as-code.yaml").toString();
+        System.out.println("config: " + config);
```

Comment:
if it is needed this println should be removed

Reply:
Looks like i got a lot to learn about the work flow. Sorry for the bad coding behaviour and i'll try to improve next time. I'm going to bed now and will modify them tomorrow. Thanks Tim!

Reply:
no worries you've done good, 
the println is useful when developing but can make test output hard to read if they're left in all over the place